src/rebuild/scan.c

```diff
@@ -633,29 +647,96 @@ placement_check(uuid_t co_uuid, vos_iter_entry_t *ent, void *data)
 	if (map != NULL)
 		pl_map_decref(map);
 
+	if (--xarg->yield_freq == 0) {
+		xarg->yield_freq = DEFAULT_YIELD_FREQ;
+		ABT_thread_yield();
+		*acts |= VOS_ITER_CB_YIELD;
```

Comment:
I think you need to track the last processed object id in the scan argument, so that you can skip it when re-probe on yield, right?

Reply:
Hmm, indeed, I should move this before the processing.